### PR TITLE
Linear Program Solver Change

### PR DESCRIPTION
**Pull request by jhkolb**
This switches the back end for our linear program solver from `lpsolve` to Apache's `SimplexSolver`.

### CODE REVIEW

File: src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala [hunk 1]
```diff
@@ -22,8 +27,9 @@ object LinearProgram {
   * @param numVars The number of variables in the linear program.
   */
 class LinearProgram(val numVars: Int) {
```
Comment (shellshocked2003) [inline src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala]: Is there a reason why there's a `val` in front of `numVars`?

**Comment by shellshocked2003** [comment]
The code looks good to me! So, is there a reason why we wouldn't prefer this over using `lpsolve`? Seems like eliminating the external dependency will make this much smoother on Amazon and we don't need any special bootstrapping anymore.

File: src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala [hunk 2]
```diff
@@ -21,9 +26,10 @@ object LinearProgram {
   * @constructor Initialize a new linear program.
   * @param numVars The number of variables in the linear program.
   */
-class LinearProgram(val numVars: Int) {
-  val solver = LpSolve.makeLp(0, numVars)
-  solver.setVerbose(LpSolve.IMPORTANT)
+class LinearProgram(numVars: Int) {
```
Comment (jhkolb) [inline src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala:29]: We don't strictly need `val` in front of `numVars`, so I have removed it. I actually had to look up some of the precise rules for constructor parameters. See here for more info if you're interested: http://stackoverflow.com/questions/14694712/do-scala-constructor-parameters-default-to-private-val

**Comment by jhkolb** [comment]
I think we initially chose `lpsolve` because we wanted to be sure we had something that was "Industrial Strength." I guess that would be an example of premature optimization. :)

It's also possible I just missed the Apache simplex solver when we first wrote the LP code.

File: src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala (hunk 2 above)
Comment (shellshocked2003) [reply src/main/scala/edu/berkeley/ce/sparkrocks/LinearProgram.scala:29]: Cool, that makes sense. Hadn't really thought about this at all until I saw the `val` in there; just assumed it never was necessary...

**Comment by shellshocked2003** [comment]
Well, luckily the setup is pretty modular since you were forward-thinking in making it easy to switch between LP solvers :) In your defense, you did mention the `breeze` solvers as a possible alternative from the outset, so I think you have done your due diligence. 

I'll merge this now!

**Pull request merged**